src/rum/instrumentation/DdRumReactNavigationTracking.tsx

```diff
@@ -30,6 +38,24 @@ export default class DdRumReactNavigationTracking {
             navigationRef.addListener("state", listener);
             this.registeredContainers.add(navigationRef);
         }
+
+        if (this.appStateListener == null) {
+            this.appStateListener = (appStateStatus: AppStateStatus) => {
+                const lastSeenViewKey = this.lastSeenRoute?.key;
+                const lastSeenViewName = this.lastSeenRoute?.name;
+                if (lastSeenViewKey != null) {
```

Comment:
Based on my own test, it seems that you shouldn't have nested `NavigationContainer`. There's still an option to do that but IMHO, it should not be that common. Let's use the fact that there's only one `NavigationContainer` per app as our base axiom, update our tracking code accordingly, and document this in the Readme.
We can then instead of a `WeakSet` simply keep a reference to the "last known `navigationContainerRef`". That way even if the nav ref is recreated in some way, we will replace it and discard the old one. 
<img width="404" alt="Screenshot 2021-03-24 at 09 17 12" src="https://user-images.githubusercontent.com/818706/112277379-fd1ae480-8c81-11eb-8610-58656e9a250a.png">

Reply:
I am (personally) curious to know if some people have multiple react entries. One single JS runtime but different entries/"mount" on the same screen . For example the top bar and bottom bar are 2 mounted RN entries (but not nested into a parent entry); while the content is some native android.